### PR TITLE
Lasb 741

### PR DESCRIPTION
**Pull request by Mhowell494**
## What

[Link to story](https://dsdmoj.atlassian.net/browse/LASB-741)

Describe what you did and why.

- Financial assessments details can now be persisted alongside assessments.
- The structure of the existing code has been improved. Business logic now resides in the service layer.

## Checklist

Before you ask people to review this PR:

- [x] Tests should be passing: `./gradlew test`
- [x] Github should not be reporting conflicts; you should have recently run `git rebase master`.
- [x] Avoid mixing whitespace changes with code changes in the same commit. These make diffs harder to read and conflicts more likely.
- [x] You should have looked at the diff against master and ensured that nothing unexpected is included in your changes.
- [x] You should have checked that the commit messages say why the change was made.

### CODE REVIEW

File: maat-court-data-api/src/main/java/gov/uk/courtdata/assessment/impl/FinancialAssessmentImpl.java [hunk 1]
```diff
@@ -1,56 +1,83 @@
...
+    public FinancialAssessmentEntity find(Integer financialAssessmentId) {
         return financialAssessmentRepository.getById(financialAssessmentId);
     }
 
-    public FinancialAssessmentEntity updateAssessment(FinancialAssessmentEntity financialAssessment) {
-        if (financialAssessment.getFullAssessmentDate() != null) {
-            financialAssessment.setAssessmentType(FinancialAssessmentType.FULL.getValue());
+    public FinancialAssessmentEntity update(FinancialAssessmentDTO financialAssessment) {
+        FinancialAssessmentEntity existingAssessment = financialAssessmentRepository.getById(financialAssessment.getId());
+        existingAssessment.setFassInitStatus(financialAssessment.getFassInitStatus());
```
Comment (atharmajeed) [inline maat-court-data-api/src/main/java/gov/uk/courtdata/assessment/impl/FinancialAssessmentImpl.java:30]: any specific reason for now using builder here?

**Review by atharmajeed — COMMENTED** [review]
Overall looks good and just added a few small comments.

File: maat-court-data-api/src/main/java/gov/uk/courtdata/assessment/impl/FinancialAssessmentImpl.java (hunk 1 above)
Comment (Mhowell494) [reply maat-court-data-api/src/main/java/gov/uk/courtdata/assessment/impl/FinancialAssessmentImpl.java:30]: Unfortunately, using a mapper here will not work - the null values in the new object don't get updated after saving due to limitations in the EntityManager merge method which gets called when executing a save operation when there is an existing record.

File: maat-court-data-api/src/main/java/gov/uk/courtdata/dto/FinancialAssessmentDTO.java [hunk 2]
```diff
@@ -25,8 +28,8 @@
     private LocalDateTime initialAssessmentDate;
     private String initOtherBenefitNote;
     private String initOtherIncomeNote;
-    private Double initTotAggregatedIncome;
-    private Double initAdjustedIncomeValue;
+    private BigDecimal initTotAggregatedIncome;
```
[the remark this answers is not in the document]
Comment (Mhowell494) [reply maat-court-data-api/src/main/java/gov/uk/courtdata/dto/FinancialAssessmentDTO.java:31]: It's mostly out of an abundance of caution. I've noticed that the MAAT application is using BigDecimal for these fields, so I thought it better to remain consistent.

**Review by atharmajeed — APPROVED** [review]
thanks for the comments and happy to approve the pr.

**Pull request merged**